src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -229,6 +230,11 @@ internal CookieContainer CookieContainer
             set
             {
                 CheckDisposedOrStarted();
+                if (value == null)
+                {
+                    throw new ArgumentNullException("CookieContainer", SR.net_http_invalid_cookiecontainer);
```

Comment:
You should NOT throw an exception in the setter for CookieContainer property.  The property can be NULL but is only valid if 'UseCookies' is FALSE.  We don't have ordering dependencies in property setters as this would confuse users.  So, for some features like cookies that depend on combinations of properties (UseCookies, CookieContainer), we validate the combination in the SendAsync() method when sending a request.  If the combination is invalid then we throw InvalidOperationException.  You can see this logic in the WinHttpHandler code.